cw_bitcoin/lib/bitcoin_wallet.dart

```diff
@@ -143,8 +144,20 @@ abstract class BitcoinWalletBase extends ElectrumWallet with Store {
     final network = walletInfo.network != null
         ? BasedUtxoNetwork.fromName(walletInfo.network!)
         : BitcoinNetwork.mainnet;
+
     final snp = await ElectrumWalletSnapshot.load(name, walletInfo.type, password, network);
 
+    final WalletKeysData keysData;
+    // Migrate wallet from the old scheme to then new .keys file scheme
+    if (!(await WalletKeysFile.hasKeysFile(name, walletInfo.type))) {
+      final newKeysData =
+          WalletKeysData(mnemonic: snp.mnemonic, xPub: snp.xpub, passphrase: snp.passphrase);
+      await WalletKeysFile.createKeysFile(name, walletInfo.type, password, newKeysData);
+      keysData = newKeysData;
+    } else {
+      keysData = await WalletKeysFile.readKeysFile(name, walletInfo.type, password);
+    }
```

Comment:
like we discussed, this should not be awaited, and preferably in a separate isolate

and btw, like this you are still depending on the snapshot file, so if it gets corrupted, the code won't reach this stage anyway